### PR TITLE
[pt2] add meta function for `take`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #98451

### CODE REVIEW

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -49,6 +49,45 @@ def toRealValueType(dtype):
     return from_complex.get(dtype, dtype)
 
 
+@register_meta([aten.take.default, aten.take.out])
+def meta_take(self, index, *, out=None):
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:53]: Same checks as in `take_out` in core.

File: torch/_meta_registrations.py [hunk 2]
```diff
@@ -49,6 +49,45 @@ def toRealValueType(dtype):
...
+        not (self.numel() == 0 and index.numel() != 0),
+        lambda: "take(): tried to take from an empty tensor",
+        IndexError,
+    )
+
+    result = self.new_empty(index.shape)
+    if out is not None:
+        assert isinstance(out, TensorLike)
+        out = _maybe_resize_out(out, result.shape)
+        return _safe_copy_out(copy_from=result, copy_to=out)  # type: ignore[arg-type]
```
Comment (nkaretnikov) [inline torch/_meta_registrations.py:87]: `out` is handled without `out_wrapper` to support these, similar to the `mean` ref.

Comment (ezyang) [reply torch/_meta_registrations.py:87]: Hmm, maybe we should make another wrapper for this, or maybe we need to modify out_wrapper?

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**